haystack/preview/embedding_backends/sentence_transformers_backend.py

```diff
@@ -0,0 +1,40 @@
+from typing import List, Optional, Union, Dict
+import hashlib
+import numpy as np
+
+from haystack.lazy_imports import LazyImport
+
+with LazyImport(message="Run 'pip install farm-haystack[inference]'") as sentence_transformers_import:
+    from sentence_transformers import SentenceTransformer
+
+
+class SentenceTransformersEmbeddingBackend:
```

Comment:
Partly inspired by https://github.com/deepset-ai/haystack/blob/c26f1e94261c298ae6f500123ff23811a60100b9/haystack/nodes/retriever/_embedding_encoder.py#L119

Reply:
Is this class meant to be part of public API?

Reply:
No.
Do you suggest to name it `_SentenceTransformersEmbeddingBackend`?

Reply:
Yep, let's make it clear it's not meant to be public.